### ISSUE TITLE
New UI, feedback and suggestions

### ISSUE THREAD

**Issue by lloesche** [mesosphere/marathon#2612]
I love the new UI but working with it I've stumbled over a couple of impracticalities that I think could be easily fixed:

1) I find myself often clicking into folders that have only one item in them, mostly another folder followed by an app. It would be nice if folders with just one item in them would get collapsed (like github does).

2) I can no longer use my browsers ctrl+f to find an app. Instead I have to use the Filter list. However when searching for something in the Filter list if the item is found in a subfolder I only see the parent folder name. And if there's multiple folders matching my search string I have to manually click through all of them to find the app I'm looking for.
2 a) it would be nice if the UI would show the matching item inside/underneath the parent folder. Like an auto-expand that shows only the matching item
2 b) it would be nice if instead of having to click into the Filter list field I could just start typing to start my search (like myspace.com)

**Comment by leemunroe** [comment]
@lloesche Great feedback. Here's an option we're exploring. Would like to hear your thoughts on whether you think this solves the problems you outlined.

1. When searching or filtering you see a list of applications with full ID (no folders)
1. Searching is always a new global search
1. Filters are always applied to the current list of applications (which could be search results, root list or folder list)

https://invis.io/BF4VSTXA5

**Comment by pierlo-upitup** [comment]
+1 and I also vote for adding a global keyboard shortcut to focus on the search field

**Comment by pierlo-upitup** [comment]
@lloesche regarding 2b, I didn't notice that we actually have the global `s` key shortcut in 0.13 https://github.com/mesosphere/marathon-ui/commit/dd364bb69b241dbf9f81e430e52b26501e90beb1!

**Comment by leemunroe** [comment]
@pierlo-upitup Didn't realize that either. Would it make sense to always autofocus this field when user lands on `/apps`?

**Comment by pierlo-upitup** [comment]
Yeah, I like that.

**Comment by pierlo-upitup** [comment]
But maybe only once we have Global search in place.

**Comment by lloesche** [comment]
@leemunroe I like the way it's visualised in the mockup. However I'm not too sure how to communicate to the user that sometimes he sees folders and sometimes he doesn't. I'd somehow like for the folder icon to not disappear and just auto-expand folders and (some) of their content if that makes sense. Sort of like a filtered tree view.

Also I think it would be nice to use some kind of heuristic that depending on the "amount of stuff" to show determines whether or not to auto-expand folders.

For instance, if I only have 4 apps in 4 folders I'd likely find folders useless regardless. Similarly if I have a large system with 5k apps that all happen to have the letter 'L' in their name I wouldn't want search to stop showing folders when I hit the 'L' key and render several pages with 5k apps (I'm also not be a fan of workarounds like "only start searching with 3+ letters").

**Comment by pierlo-upitup** [comment]
I think most of the problems raised in this issues are covered by the global search story ( https://github.com/mesosphere/marathon/issues/2644 ) and have been merged into current master for the upcoming 0.13 release. 
Please take a look at the recently merged PR where the breadcrumb are shown under the app's name : https://github.com/mesosphere/marathon-ui/pull/436 for a quick check of how it works.

Additionally, a Tree View is an already planned feature, so I think we can close this issue for now. Feel free to re-open otherwise!
